In [ ]:
import numpy as np
import pandas as pd

#### Question 1: Read in the surveys.csv dataset and call it surveys_df.

In [3]:
import pandas as pd
surveys_df = pd.read_csv('surveys.csv')
surveys_df

record_id  month  day  year  plot_id species_id  sex  hindfoot_length  \
0              1      7   16  1977        2         NL    M             32.0   
1              2      7   16  1977        3         NL    M             33.0   
2              3      7   16  1977        2         DM    F             37.0   
3              4      7   16  1977        7         DM    M             36.0   
4              5      7   16  1977        3         DM    M             35.0   
...          ...    ...  ...   ...      ...        ...  ...              ...   
35544      35545     12   31  2002       15         AH  NaN              NaN   
35545      35546     12   31  2002       15         AH  NaN              NaN   
35546      35547     12   31  2002       10         RM    F             15.0   
35547      35548     12   31  2002        7         DO    M             36.0   
35548      35549     12   31  2002        5        NaN  NaN              NaN   

       weight  
0         NaN  
1         NaN  
2         NaN  
3         NaN  
4         NaN  
...       ...  
35544     NaN  
35545     NaN  
35546    14.0  
35547    51.0  
35548     NaN  

[35549 rows x 9 columns]

#### Question 2: How many rows and columns are in the dataset?

In [17]:
rows, columns=surveys_df.shape

print("Number of rows = ", rows)
print("Number of columns = ", columns)

Number of rows =  35549
Number of columns =  9


#### Question 3: Use the head() function to displays the first 10 lines in the dataset.

In [4]:
surveys_df.head(10)

record_id  month  day  year  plot_id species_id sex  hindfoot_length  \
0          1      7   16  1977        2         NL   M             32.0   
1          2      7   16  1977        3         NL   M             33.0   
2          3      7   16  1977        2         DM   F             37.0   
3          4      7   16  1977        7         DM   M             36.0   
4          5      7   16  1977        3         DM   M             35.0   
5          6      7   16  1977        1         PF   M             14.0   
6          7      7   16  1977        2         PE   F              NaN   
7          8      7   16  1977        1         DM   M             37.0   
8          9      7   16  1977        1         DM   F             34.0   
9         10      7   16  1977        6         PF   F             20.0   

   weight  
0     NaN  
1     NaN  
2     NaN  
3     NaN  
4     NaN  
5     NaN  
6     NaN  
7     NaN  
8     NaN  
9     NaN

#### Question 4: Use the info() function to display the datatypes. How many of the variables are categorical and how many are quantitative?

In [6]:
surveys_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35549 entries, 0 to 35548
Data columns (total 9 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   record_id        35549 non-null  int64  
 1   month            35549 non-null  int64  
 2   day              35549 non-null  int64  
 3   year             35549 non-null  int64  
 4   plot_id          35549 non-null  int64  
 5   species_id       34786 non-null  object 
 6   sex              33038 non-null  object 
 7   hindfoot_length  31438 non-null  float64
 8   weight           32283 non-null  float64
dtypes: float64(2), int64(5), object(2)
memory usage: 2.4+ MB


 There are 7 quantitative variables i.e. record_id, month, day, year, plot_id, hindfoot_length, and weight whereas the remaining 2 variables are categorical i.e. species_id.

#### Question 5: Print out the column names.

In [8]:
surveys_df.columns

Index(['record_id', 'month', 'day', 'year', 'plot_id', 'species_id', 'sex',
       'hindfoot_length', 'weight'],
      dtype='object')

#### Question 6: Use the unique() function to obtain the list of all the species.

In [20]:
pd.unique(surveys_df['species_id'])
surveys_df.species_id.value_counts()

DM    10596
PP     3123
DO     3027
PB     2891
RM     2609
DS     2504
OT     2249
PF     1597
PE     1299
NL     1252
OL     1006
PM      899
AH      437
AB      303
SS      248
SH      147
SA       75
RF       75
CB       50
BA       46
SO       43
SF       43
DX       40
PC       39
PL       36
PH       32
CQ       16
CM       13
OX       12
UR       10
PI        9
UP        8
PG        8
RO        8
PX        6
SU        5
PU        5
US        4
UL        4
AS        2
RX        2
ZL        2
ST        1
CU        1
SC        1
CS        1
CT        1
CV        1
Name: species_id, dtype: int64

#### Question 7: Obtain the descriptive statistics of all the numerical variables in the dataframe.

In [13]:
surveys_df.describe()

record_id         month           day          year       plot_id  \
count  35549.000000  35549.000000  35549.000000  35549.000000  35549.000000   
mean   17775.000000      6.474022     16.105966   1990.475231     11.397001   
std    10262.256696      3.396583      8.256691      7.493355      6.799406   
min        1.000000      1.000000      1.000000   1977.000000      1.000000   
25%     8888.000000      4.000000      9.000000   1984.000000      5.000000   
50%    17775.000000      6.000000     16.000000   1990.000000     11.000000   
75%    26662.000000      9.000000     23.000000   1997.000000     17.000000   
max    35549.000000     12.000000     31.000000   2002.000000     24.000000   

       hindfoot_length        weight  
count     31438.000000  32283.000000  
mean         29.287932     42.672428  
std           9.564759     36.631259  
min           2.000000      4.000000  
25%          21.000000     20.000000  
50%          32.000000     37.000000  
75%          36.000000     48.000000  
max          70.000000    280.000000

* The mean, max, and min of the hindfoot_length are: 29.287932, 70.000000	, 2.000000 respectively.
* The mean, max, and min of the hindfoot_length are: 42.672428, 280.000000	, 4.000000 respectively.



#### Question 8: Using the groupby() function, group the dataset by sex and compute the descriptive statistics.

In [14]:
surveys_df.groupby("sex").describe()

record_id                                                               \
        count          mean           std  min      25%      50%       75%   
sex                                                                          
F     15690.0  18036.412046  10423.089000  3.0  8917.50  18075.5  27250.00   
M     17348.0  17754.835601  10132.203323  1.0  8969.75  17727.5  26454.25   

                month            ... hindfoot_length         weight  \
         max    count      mean  ...             75%   max    count   
sex                              ...                                  
F    35547.0  15690.0  6.583047  ...            36.0  64.0  15303.0   
M    35548.0  17348.0  6.392668  ...            36.0  58.0  16879.0   

                                                         
          mean        std  min   25%   50%   75%    max  
sex                                                      
F    42.170555  36.847958  4.0  20.0  34.0  46.0  274.0  
M    42.995379  36.184981  4.0  20.0  39.0  49.0  280.0  

[2 rows x 56 columns]

#### Question 9: Using the groupby() function, group the dataset by species_id and compute the descriptive statistics.

In [15]:
surveys_df.groupby("species_id").describe()

record_id                                                          \
               count          mean           std      min       25%      50%   
species_id                                                                     
AB             303.0  13201.702970   5835.874869   3126.0   7604.00  13829.0   
AH             437.0  21631.716247   7194.381201   1446.0  15507.00  21600.0   
AS               2.0  19760.000000   1170.968830  18932.0  19346.00  19760.0   
BA              46.0  18586.043478    921.349203  16598.0  17972.50  18803.0   
CB              50.0  18300.460000   7271.702586   3189.0  13227.00  19590.5   
CM              13.0   4055.000000   2789.266660   3190.0   3263.00   3291.0   
CQ              16.0   5718.062500   1899.301607   3200.0   3720.75   6199.0   
CS               1.0  15878.000000           NaN  15878.0  15878.00  15878.0   
CT               1.0  28988.000000           NaN  28988.0  28988.00  28988.0   
CU               1.0  21209.000000           NaN  21209.0  21209.00  21209.0   
CV               1.0  13313.000000           NaN  13313.0  13313.00  13313.0   
DM           10596.0  15776.228011   9758.042325      3.0   7466.75  15092.5   
DO            3027.0  17970.353485   9529.235809     68.0  10263.50  17078.0   
DS            2504.0   6096.122204   5002.986094     11.0   2309.25   4727.5   
DX              40.0  22236.350000   6588.027227   2706.0  19710.50  20573.5   
NL            1252.0  13902.781949   9459.788377      1.0   6433.00  12609.0   
OL            1006.0  11097.006958   6911.769060    274.0   5414.50  10230.5   
OT            2249.0  19908.781236  10566.087720     36.0  10438.00  21535.0   
OX              12.0   4548.666667   7204.725739     70.0    174.25    287.5   
PB            2891.0  31423.223452   2822.580268  22711.0  29478.00  31659.0   
PC              39.0  17035.974359   4799.825751   6567.0  14617.00  18500.0   
PE            1299.0  15920.995381   8076.090716      7.0  11336.50  15107.0   
PF            1597.0  17347.086412   9151.113902      6.0   7210.00  21566.0   
PG               8.0  16909.000000   6389.064385   7352.0  14992.75  17017.0   
PH              32.0  16061.343750   4843.685258   9476.0  10943.00  17213.5   
PI               9.0  31139.444444   5558.186915  20408.0  32708.00  33769.0   
PL              36.0  26021.750000   1856.592507  22817.0  24864.75  26629.0   
PM             899.0  19354.033370   8470.264362    883.0  12496.50  23878.0   
PP            3123.0  25281.530900   7599.657786     18.0  21748.50  26181.0   
PU               5.0  16813.000000   6548.811304   9643.0   9650.00  21342.0   
PX               6.0  26337.333333   2903.952663  23435.0  24749.50  25064.0   
RF              75.0  15888.293333   2076.856856   5764.0  15480.00  15929.0   
RM            2609.0  16410.623994   6241.789011    311.0  13027.00  16752.0   
RO               8.0  32526.875000   5589.994823  18871.0  33516.25  34041.0   
RX               2.0  23703.500000   2006.061938  22285.0  22994.25  23703.5   
SA              75.0   9089.466667   5389.253424    779.0   5743.00   8314.0   
SC               1.0  16002.000000           NaN  16002.0  16002.00  16002.0   
SF              43.0  20809.116279   6630.187048  16659.0  16994.00  17514.0   
SH             147.0  19330.231293   7796.076966     24.0  15046.50  16088.0   
SO              43.0  20457.627907   1616.424096  19125.0  19552.50  19972.0   
SS             248.0  16291.798387   9037.762489     77.0   9436.75  14832.0   
ST               1.0  20723.000000           NaN  20723.0  20723.00  20723.0   
SU               5.0  22160.400000    197.865611  21807.0  22233.00  22247.0   
UL               4.0  19403.250000   3381.668954  14466.0  18580.50  20773.0   
UP               8.0  14816.750000   1327.000565  13688.0  14013.25  14226.5   
UR              10.0  21367.000000   8568.125738  13564.0  13719.75  19103.5   
US               4.0  35524.250000     15.063753  35512.0  35512.75  35520.5   
ZL         